src/txdb.cpp

```diff
@@ -146,8 +146,7 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
             changed++;
         }
         count++;
-        CCoinsMap::iterator itOld = it++;
-        mapCoins.erase(itOld);
+        it = erase ? mapCoins.erase(it) : std::next(it);
```

Comment:
In the first commit 79cedc36afe2, I'm unsure if this diff is a change of behavior, though haven't looked more into https://github.com/bitcoin/bitcoin/pull/17487#discussion_r1082702771 and if it applies here.

Reply:
Thanks @Sjors, agree, as a refresher I just verified this with https://godbolt.org/z/h7Ex9dqrK

<details><summary>code</summary><p>

```cpp
#include <iostream>
#include <iterator>
#include <string>
#include <unordered_map>

int main()
{
    auto print_key_value = [](const auto& key, const auto& value)
    {
        std::cout << "Key:[" << key << "] Value:[" << value << "]\n";
    };

    std::unordered_map<std::string, std::string> m {
        {"RED","#FF0000"},
        {"GREEN","#00FF00"},
        {"BLUE","#0000FF"},
        {"BLACK", "#000000"},
        {"WHITE", "#FFFFFF"},
    };

    std::cout << "Iterate on unordered_map using std::next\n";
    for (auto it = m.begin(); it != m.end();) {
        print_key_value(it->first, it->second);
        it = std::next(it);
    }

    std::cout << "\nSame, using increment iterator and erase\n";
    for (auto it = m.begin(); it != m.end();) {
        print_key_value(it->first, it->second);
        auto itOld = it++;
        m.erase(itOld);
    }

    if (m.empty()) {
        std::cout << "\nUnordered_map is now empty\n";
    }

    return 0;
}
```

</p></details>

<details><summary>output</summary><p>

```
Iterate on unordered_map using std::next
Key:[BLACK] Value:[#000000]
Key:[BLUE] Value:[#0000FF]
Key:[WHITE] Value:[#FFFFFF]
Key:[GREEN] Value:[#00FF00]
Key:[RED] Value:[#FF0000]

Same, using increment iterator and erase
Key:[BLACK] Value:[#000000]
Key:[BLUE] Value:[#0000FF]
Key:[WHITE] Value:[#FFFFFF]
Key:[GREEN] Value:[#00FF00]
Key:[RED] Value:[#FF0000]

Unordered_map is now empty
```

</p></details>